### PR TITLE
Direction changes.

### PR DESCRIPTION
**Pull request by Ziemas**
According to Necrolis the game uses 64 directions for entities which then get mapped down to however many directions a specific asset has by way of a lookup table.

### CODE REVIEW

File: d2core/d2asset/animation.go [hunk 1]
```diff
@@ -251,12 +251,9 @@ func (a *Animation) GetDirectionCount() int {
 }
 
 func (a *Animation) SetDirection(directionIndex int) error {
-	if directionIndex >= len(a.directions) {
-		return errors.New("invalid direction index")
-	}
-
-	a.directionIndex = directionIndex
+	a.directionIndex = d2dcc.Dir64ToDcc(directionIndex, len(a.directions))
```
Comment (nicholas-eden) [inline d2core/d2asset/animation.go]: Should we remove the error return?  Or panic if directionIndex >= 64?

File: d2core/d2asset/composite.go [hunk 2]
```diff
@@ -126,8 +127,12 @@ func (c *Composite) createMode(animationMode, weaponClass string, direction int)
 		return nil, err
 	}
 
-	if direction >= cof.NumberOfDirections {
-		return nil, errors.New("invalid direction")
+	// oh god how do i math
+	offset := (64 / cof.NumberOfDirections) / 2
+	entityDirection := int(math.Trunc((float64(direction+offset)-64.0)*(-float64(cof.NumberOfDirections)/-64.0) + float64(cof.NumberOfDirections)))
```
Comment (nicholas-eden) [inline d2core/d2asset/composite.go]: It looks like `math.Trunc` is redundant
```suggestion
	int((float64(direction+offset)-64.0)*(-float64(cof.NumberOfDirections)/-64.0) + float64(cof.NumberOfDirections))
```

File: d2core/d2map/animated_entity.go [hunk 3]
```diff
@@ -42,21 +41,9 @@ func (ae AnimatedEntity) GetDirection() int {
...
-	case 8:
-		layerDirection = d2dcc.CofToDir8[ae.direction]
-	case 16:
-		layerDirection = d2dcc.CofToDir16[ae.direction]
-	case 32:
-		layerDirection = d2dcc.CofToDir32[ae.direction]
-	}
-
-	ae.animation.SetDirection(layerDirection)
+	ae.animation.SetDirection(ae.direction)
```
Comment (nicholas-eden) [inline d2core/d2map/animated_entity.go]: This is so much better

File: d2core/d2asset/animation.go (hunk 1 above)
Comment (Ziemas) [reply d2core/d2asset/animation.go]: Added the check back with >= 64

**Pull request merged**